pkg/driver/node/mounter/systemd_mounter.go

```diff
@@ -85,25 +88,21 @@ func (m *SystemdMounter) Mount(ctx context.Context, bucketName string, target st
 					}
 				}
 			}()
-		}
-		// Corrupted mount, try unmounting
-		if mount.IsCorruptedMnt(mountpointErr) {
+			// Corrupted mount, try unmounting
+		} else if m.MpMounter.IsMountPointCorrupted(err) {
 			klog.V(4).Infof("Mount: Target path %q is a corrupted mount. Trying to unmount.", target)
 			if mntErr := m.Unmount(ctx, target, credentialprovider.CleanupContext{
 				WritePath: credentialCtx.WritePath,
 				PodID:     credentialCtx.PodID,
 				VolumeID:  credentialCtx.VolumeID,
 			}); mntErr != nil {
-				return fmt.Errorf("Unable to unmount the target %q : %v, %v", target, mountpointErr, mntErr)
+				return fmt.Errorf("Unable to unmount the target %q : %v, %v", target, err, mntErr)
 			}
+		} else {
+			return fmt.Errorf("Could not check if %q is a mount point: %v, %v", target, err, err)
```

Comment:
We are printing the same error twice here?

Reply:
I think it looks like that due to diff, I double-checked the file, and it doesn't seem to be the case https://github.com/awslabs/mountpoint-s3-csi-driver/blob/0bdc09aaed2a3954a7c91bd2ab992db6c6c15957/pkg/driver/node/mounter/systemd_mounter.go#L75-L104

Reply:
Sorry misunderstood, fixed with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/444/commits/48e3115c97e7310189af2a9bcbe1616de10d7885